docs/architecture/mobile-clients/watchOS.mdx

```diff
@@ -0,0 +1,184 @@
+- Services (totp generation, crypto services and business logic)
+- Presentation (use `SwiftUI` for the UI with an MVVM pattern)
+
+## Integration with iOS
+
+The watchOS app is developed using `XCode` and `Swift` and we need to integrate it to the `Xamarin`
+iOS application.
+
+For this, the `iOS.csproj` has been adapted taking a
+[solution](https://github.com/xamarin/xamarin-macios/issues/10070#issuecomment-1033428823) provided
+in the `Xamarin.Forms` GitHub repository and modified to our needs:
+
+```xml
+<PropertyGroup>
+    <WatchAppBuildPath Condition=" '$(Configuration)' == 'Debug' ">$(Home)/Library/Developer/Xcode/DerivedData/bitwarden-cbtqsueryycvflfzbsoteofskiyr/Build/Products</WatchAppBuildPath>
+    <WatchAppBuildPath Condition=" '$(Configuration)' != 'Debug' ">$([System.IO.Path]::GetFullPath('$(MSBuildProjectDirectory)\..'))/watchOS/bitwarden.xcarchive/Products/Applications/bitwarden.app/Watch</WatchAppBuildPath>
+    <WatchAppBundle>Bitwarden.app</WatchAppBundle>
+    <WatchAppConfiguration Condition=" '$(Platform)' == 'iPhoneSimulator' ">watchsimulator</WatchAppConfiguration>
+    <WatchAppConfiguration Condition=" '$(Platform)' == 'iPhone' ">watchos</WatchAppConfiguration>
+    <WatchAppBundleFullPath Condition=" '$(Configuration)' == 'Debug' ">$(WatchAppBuildPath)/$(Configuration)-$(WatchAppConfiguration)/$(WatchAppBundle)</WatchAppBundleFullPath>
+    <WatchAppBundleFullPath Condition=" '$(Configuration)' != 'Debug' ">$(WatchAppBuildPath)/$(WatchAppBundle)</WatchAppBundleFullPath>
+</PropertyGroup>
+
+...
+
+<ItemGroup Condition=" '$(Configuration)' == 'Debug' AND Exists('$(WatchAppBundleFullPath)') ">
+    <_ResolvedWatchAppReferences Include="$(WatchAppBundleFullPath)" />
+</ItemGroup>
+<ItemGroup Condition=" '$(Configuration)' != 'Debug' ">
+    <_ResolvedWatchAppReferences Include="$(WatchAppBundleFullPath)" />
+</ItemGroup>
+<PropertyGroup Condition=" '$(_ResolvedWatchAppReferences)' != '' ">
+    <CodesignExtraArgs>--deep</CodesignExtraArgs>
+</PropertyGroup>
+<Target Name="PrintWatchAppBundleStatus" BeforeTargets="Build">
+    <Message Text="WatchAppBundleFullPath: '$(WatchAppBundleFullPath)' exists" Condition=" Exists('$(WatchAppBundleFullPath)') " />
+    <Message Text="WatchAppBundleFullPath: '$(WatchAppBundleFullPath)' does NOT exist" Condition=" !Exists('$(WatchAppBundleFullPath)') " />
+</Target>
+```
+
+So on the `PropertyGroup` the `WatchAppBundleFullPath` is assembled together depending on the
+Configuration and the Platform taking the output of the XCode watchOS app build. Then there are some
+`ItemGroup` to include the watch app depending on if it exists and the Configuration. The task
+`_ResolvedWatchAppReferences` is the one responsible to peek into the `Bitwarden.app` built by XCode
+and if it finds a Watch app, it will just bundle it to the Xamarin iOS application. Finally, if the
+Watch app is bundled deep signing is enabled and on building the build path is printed if exists.
```

Comment:
> Finally, if the
Watch app is bundled deep signing is enabled and on building the build path is printed if exists.

Think there are some commas missing.